server/src/test/java/io/crate/replication/logical/action/TransportCreateSubscriptionActionTest.java

```diff
@@ -83,12 +92,28 @@ private void subscribeToTablesWithVersion(FutureActionListener<AcknowledgedRespo
             mock(TransportService.class),
             clusterService,
             logicalReplicationService,
-            THREAD_POOL,
+            mock(ThreadPool.class),
             userLookup
         );
 
         when(userLookup.findUser(anyString())).thenReturn(User.CRATE_USER);
 
+        final DiscoveryNode dataNode = new DiscoveryNode(
+            "node",
+            buildNewFakeTransportAddress(),
+            Map.of(),
+            Set.of(),
+            Version.CURRENT
+        );
+        ClusterState clusterState = ClusterState
+            .builder(ClusterName.DEFAULT)
+            .nodes(DiscoveryNodes.builder().add(dataNode).localNodeId(dataNode.getId()))
+            .build();
+        when(clusterService.state()).thenReturn(clusterState);
+
+        doAnswer(invocation -> responseFuture.complete(new AcknowledgedResponse(true)))
+            .when(clusterService).submitStateUpdateTask(anyString(), any());
```

Comment:
Why this change, how is it related?

Reply:
Previously, non-mock `clusterService`, inherited from `CrateDummyClusterServiceUnitTest` used to work when submitting regular tasks. `clusterStateProcessed() ` is hit and future is completed.

`onAllNodesAcked` is never hit in unit tests - I think because we don't have any other nodes to ack? 

Because of that I made clusterService mock and then had to mock clusterService.state() as well.